### PR TITLE
Bug 1473356 - Script the creation of a WNP blob (schema 9)

### PR DESCRIPTION
**Pull request by JohanLorenzo**
~~Taskcluster fails because mozilla-version 0.2.0 isn't released yet. Depends on https://github.com/mozilla-releng/mozilla-version/pull/5~~ Fixed!

### CODE REVIEW

**Comment by JohanLorenzo** [comment]
I let you thumbwrestle after you guys are back :)

**Comment by MihaiTabara** [comment]
Once this is reviewed and tested and working, I think it's worth mentioning this in the https://github.com/mozilla-releng/releasewarrior-2.0/blob/master/docs/CHANGELOG.md as well 👍

**Comment by lundjordan** [comment]
sorry for delay here, it's on me. I told Tom I'd get to it :)

File: releasewarrior/balrog.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    url = '{}/releases/{}'.format(BALROG_API_ROOT, blob_name)
+    req = requests.get(url, verify=True, timeout=4)
+    req.raise_for_status()
+    return req.json()
+
+
+def get_releases(blob_name, name_prefix=None):
+    url = '{}/releases'.format(BALROG_API_ROOT)
+    params = {
+        'product': extract_product_from_blob_name(blob_name),
```
Comment (lundjordan) [inline releasewarrior/balrog.py]: nice

File: releasewarrior/balrog.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        if _ENDS_WITH_BUILD_REGEX.search(release)
+    ]
+    balrog_releases = [BalrogReleaseName.parse(release) for release in ends_with_build]
+
+    same_type = [
+        release
+        for release in balrog_releases
+        if release.version.version_type == original_release.version.version_type
+    ]
+    if original_release.version.is_release:
```
Comment (lundjordan) [inline releasewarrior/balrog.py]: nice catch

File: releasewarrior/balrog.py [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+        previous_release = find_previous_release(blob_name, all_releases_names_for_product)
+        previous_release_blob = get_release_blob(blob_name=previous_release)
+        for_locales = _get_locales_from_blob(previous_release_blob, previous_release)
+        log.info('for_locales gotten from previous "{}": {}'.format(previous_release, for_locales))
+    else:
+        for_locales = [locale.strip() for locale in for_locales.split(',')]
+        log.info('Using for_locales from command line: {}'.format(for_locales))
+
+    if not isinstance(for_locales, list):
+        raise BalrogError('{} is not a list')
```
Comment (lundjordan) [inline releasewarrior/balrog.py]: missing format?

File: releasewarrior/cli.py [hunk 4]
```diff
@@ -381,3 +388,57 @@ def data(push, pull, logger=LOGGER, config=CONFIG):
         git.push(logger, config)
     if pull:
         git.pull(logger, config)
+
+
+@cli.command()
+@click.argument('wnp_url')
+@click.argument('product', type=click.Choice(['firefox', 'devedition', 'fennec', 'thunderbird']))
```
Comment (lundjordan) [inline releasewarrior/cli.py]: Should we just have Firefox here for now?

**Comment by lundjordan** [comment]
> It would be great to evolve this further (stage 2?) and make it as a task within release graphs. iow - within automation and then even uploaded to balrog. :)

I guess part of that deps on what we decide with your great proposal here: Bug 1482395 - Proposal: Move the WNP rules in-tree

File: releasewarrior/cli.py (hunk 4 above)
Comment (JohanLorenzo) [reply releasewarrior/cli.py]: Good point. I added a comment if folks want to add support for it.

File: releasewarrior/balrog.py (hunk 3 above)
Comment (JohanLorenzo) [reply releasewarrior/balrog.py]: Good catch. Done!

**Comment by MihaiTabara** [comment]
Maybe worth adding this in the CHANGELOG to ease future releaseduty catch-up. Good job on this again.

**Pull request merged**